tsl/src/continuous_aggs/refresh.c

```diff
@@ -290,15 +291,71 @@ log_refresh_window(int elevel, const ContinuousAgg *cagg, const InternalTimeRang
 		 DatumGetCString(OidFunctionCall1(outfuncid, end_ts)));
 }
 
+#define MAX_INVALIDATIONS_PER_REFRESH 10
+
+/*
+ * Get the limit on number of invalidation-based refreshes we allow per refresh call.
+ *
+ * Use a session variable for now, for debugging and tweaking. Might promote
+ * to full GUC if useful.
+ */
+static long
+get_max_invalidations_per_refresh(void)
+{
+	const char *max_invalidations_setting =
+		GetConfigOption("timescaledb.max_invalidations_per_refresh", true, false);
+	long max_invalidations = MAX_INVALIDATIONS_PER_REFRESH;
+
+	if (max_invalidations_setting)
+	{
+		char *endptr = NULL;
+
+		/* Not using pg_strtol here since we don't want to throw error in case
+		 * of parsing issue */
+		max_invalidations = strtol(max_invalidations_setting, &endptr, 10);
```

Comment:
Why not? If the variable is set to a strange string, isn't it better to raise an error early rather than running with a default value? A user making a subtle mistake might be befuddled about why things are behaving strange.

Reply:
Can this be checked by the GUC callback or something like that? If there is none, we should validate it

Reply:
Sorry, this was not clear, that we are not going to document it

Reply:
Sorry it wasn't clear. Tried to make clear in the function: "* Use a session variable for now, for debugging and tweaking. Might promote to full GUC if useful."

Reply:
The point of raising an error here is to ensure that the value is corrected rather than using a default value, so yes, it should throw an error until the value is changed. Validating it when setting it is (of course) better.

Reply:
Added a warning as agreed.